### ISSUE TITLE
thorin::Node_TypeParam not handled in CodeGen::convert(Type type)

### ISSUE THREAD

**Issue by richardmembarth** [AnyDSL/thorin#39]
Thorin segfaults since ```thorin::Node_TypeParam``` is nod handled in ```CodeGen::convert(Type type)```, see commit AnyDSL/impala@93237d5

**Comment by leissa** [comment]
This cannot work. LLVM does not support polymorphism.

**Comment by richardmembarth** [comment]
So this needs to be resolved earlier?

**Comment by leissa** [comment]
yes, I can write a pass which rewrites all uses of polymorphic "C" externs accordingly. It's not hard.

**Comment by leissa** [comment]
We will solve this issue in the long term via traits / better type system.
